torchgeo/datasets/ssl4eo.py

```diff
@@ -0,0 +1,229 @@
+        assert split in self.metadata
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._verify()
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            sample with an "image" in 5xCxHxW format where the 5 indexes over the same
+                patch sampled from different points in time by the SeCo method
+        """
+        directory = os.path.join(self.root, self.split, f"{index // self.times:07}")
+        subdirs = sorted(os.listdir(directory))
+        directory = os.path.join(directory, subdirs[index % self.times])
+
+        images = []
+        for band in self.metadata[self.split]["bands"]:
+            filename = os.path.join(directory, f"{band}.tif")
+            with rasterio.open(filename) as f:
+                image = f.read(out_shape=(1, self.size, self.size)).astype(np.float32)
+                images.append(torch.from_numpy(image))
+
+        sample = {"image": torch.cat(images)}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return self.locations * self.times
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        directory_path = os.path.join(self.root, self.split)
+        if os.path.exists(directory_path):
+            return
+
+        # Check if the zip files have already been downloaded
+        zip_path = os.path.join(self.root, self.metadata[self.split]["filename"])
+        md5 = self.metadata[self.split]["md5"] if self.checksum else None
+        integrity = check_integrity(zip_path, md5)
+        if integrity:
+            self._extract()
+        else:
+            raise RuntimeError(f"Dataset not found in `root={self.root}`")
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        extract_archive(os.path.join(self.root, self.metadata[self.split]["filename"]))
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        if self.split == "s1":
+            # See Sentinel1.plot
```

Comment:
Wish we could call the plot methods of Sentinel1 and Sentinel2 directly so we didn't have to copy the same logic into every dataset. I bet we could figure out a way to do this...